pkg/collectsub/datasource/datasource.go

```diff
@@ -36,7 +36,17 @@ type CollectSource interface {
 
 type DataSources struct {
 	OciDataSources []Source
+	// NOTE: Git data sources should follow the URI scheme as defined in:
+	// https://spdx.github.io/spdx-spec/v2.3/package-information/#771-description
+	// <vcs_tool>+<transport>://<host_name>[/<path_to_repository>][@<revision_tag_or_branch>][#<sub_path>]
+	// e.g. git+https://github.com/guacsec/guac@v1
 	GitDataSources []Source
+	// NOTE: It is expected that a GithubReleaseDataSource is of the form:
+	// https://github.com/<org>/<repo>/releases/<tag> or
+	// https://github.com/<org>/<repo>/releases/tag/<tag> or
+	// https://github.com/<org>/<repo>/releases
+	// Tag is optional and left off will assume latest.
+	GithubReleaseDataSources []Source
```

Comment:
note: we need to update the csub interface and proto to support this. Can you add a TODO with a link to an issue and assign it to me?

Reply:
Sure.